src/logics/Logic.cc

```diff
@@ -1596,23 +1599,23 @@ bool Logic::typeCheck(SymRef sym, vec<PTRef> const & args, std::string & why) co
         Symbol const & symbol = sym_store[sym];
         if (symbol.chainable() or symbol.pairwise()) {
             for (int i = 0; i < args.size(); i++) {
-                symStr += " " + std::string(getSortName(symbol[0]));
+                symStr += " " + std::string(printSort(symbol[0]));
             }
         } else if (symbol.left_assoc()) {
-            symStr += " " + std::string(getSortName(symbol[0]));
+            symStr += " " + std::string(printSort(symbol[0]));
             for (int i = 1; i < args.size(); i++) {
-                symStr += " " + std::string(getSortName(symbol[1]));
+                symStr += " " + std::string(printSort(symbol[1]));
             }
         } else if (symbol.right_assoc()) {
             for (int i = 0; i < args.size() - 1; i++) {
-                symStr += " " + std::string(getSortName(symbol[0]));
+                symStr += " " + std::string(printSort(symbol[0]));
             }
-            symStr += " " + std::string(getSortName(symbol[1]));
+            symStr += " " + std::string(printSort(symbol[1]));
         }
 
         std::string argSorts;
         for (PTRef tr : args) {
-            argSorts += std::string(getSortName(getSortRef(tr))) + " ";
+            argSorts += std::string(printSort(getSortRef(tr))) + " ";
```

Comment:
`printSort` now already returns `string`, so we do no need to wrap it in `std::string` anymore.

Reply:
Fixed.

Reply:
There are more occurrences of this in this method.

Reply:
Ok maybe I now found most of them...